src/modules/users/users.service.ts

```diff
@@ -99,31 +100,41 @@ export class UserService {
 
     account.role = account.role === Roles.Admin ? Roles.User : Roles.Admin;
 
-    const updatedUser = await this.userAccountRepository.save(account);
+    const updatedUser = await this.accountRepository.save(account);
     return updatedUser;
   }
 
-  async deleteAccount(deleterId: string, deletingAccountId: string) {
-    const deleter = await this.userAccountRepository.findOne({
-      where: { id: deleterId, isDeleted: false },
+  async softDeleteAccount(deleteActorId: string, deletingAccountId: string) {
+    const deleteActor = await this.accountRepository.findOne({
+      where: { id: deleteActorId, isDeleted: false },
     });
-    if (deleter.role != Roles.Admin && deleterId != deletingAccountId) {
+    if (deleteActor.role != Roles.Admin && deleteActorId != deletingAccountId) {
       throw new ForbiddenException('Only owner or admin can delete account');
     }
 
-    const deletingUser = await this.userAccountRepository.findOne({
+    const deletingAccount = await this.accountRepository.findOne({
       where: { id: deletingAccountId, isDeleted: false },
     });
-    if (!deletingUser) {
+    if (!deletingAccount) {
       throw new NotFoundException('User not found or already deleted');
     }
 
-    deletingUser.isDeleted = true;
-    const deletedUser = await this.userAccountRepository.save(deletingUser);
-
-    const { ...result } = deletedUser;
-    delete result.password;
+    if (deletingAccount.role === Roles.Admin) {
+      const remainingAdmins = await this.accountRepository.count({
+        where: {
+          role: Roles.Admin,
+          isDeleted: false,
+          id: Not(deletingAccountId),
+        },
+      });
+
+      if (remainingAdmins === 0) {
```

Comment:
Maybe `remainingAdmins === 0`?

Reply:
Nope, because if you see in query I add Not(deletingAccountId) contidion, so it cannot find user itself.

But, I fixed it for making it more readable. Some job but more readable